### DISCUSSION TITLE
Nested Components and Self-closing tags consume x-data

### DISCUSSION THREAD

**Discussion by natepisarski** [alpinejs/alpine#2319] [7. Bugs (archived)] [answered]
During testing of another, unrelated bug, I stumbled upon this bug. It involves nested components, and self closing tags.

If you try this code:

```
<div x-data="{firstName: 'George1'}">
        <div x-data="{lastName: 'George2'}">
            <div x-text="firstName + lastName" />
        </div>
    </div>
```

You will see the following output:

```
George1George2
```

As you would expect. However, if you add 1 line, making the code:

```
    <div x-data="{firstName: 'George1'}">
        <div x-text="firstName"/>
        <div x-data="{lastName: 'George2'}">
            <div x-text="firstName + lastName" />
        </div>
    </div>
```

You will get the following error:

```
Alpine Expression Error: firstName is not defined

Expression: "firstName + lastName"

 
<div x-text="firstName + lastName">
```

Replacing it with this code, it works as expected again:

```
    <div x-data="{firstName: 'George1'}">
        <div x-text="firstName"></div>
        <div x-data="{lastName: 'George2'}">
            <div x-text="firstName + lastName" />
        </div>
    </div>
```

It seems like the self-closing `<div>` tag does something that causes `firstName` to be "consumed" when it's used in the x-text.

**Comment by SimoTod** [comment, marked as answer]
In html5, only void tags can be self-closing (https://dev.w3.org/html5/html-author/#self-closing-tag) but the slash is actually redundant.
You can try to build a page with this code `<html><body><div /><p></p></body></html>`, inspect the page in chrome and you'll see that the browser moved the p inside the div even if you told the browser that the tag was already closed.
These unrequested fixes and DOM shuffling probably make Alpine confuse.

**Reply by danddanddand** [reply]
I was about to write same thing, basically Alpine is at browsers mercy because it thinks it's an open div and a malformed document.
